phases/19-capstone-projects/23-function-call-dispatcher/code/main.py

```
@@ -0,0 +1,334 @@
+
+        if idempotency_key is not None:
+            now = time.monotonic()
+            cached = self._cache.get(idempotency_key)
+            if cached is not None and now - cached[1] < self._cache_ttl:
+                return DispatchOk(result=cached[0], attempts=0)
+            inflight = self._inflight.get(idempotency_key)
+            if inflight is not None:
+                try:
+                    res = await inflight.future
+                    return DispatchOk(result=res, attempts=0)
+                except Exception as exc:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)